### PR TITLE
feat: Expose Org Browser for Scratch Orgs

### PR DESCRIPTION
**Pull request by klewis-sfdc**
### What does this PR do?
* Enables the Org Browser feature to be used with scratch orgs.

### What issues does this PR fix or reference?
@W-10931259@

### Functionality Before
* Org Browser was not available when connected to a scratch org.

### Functionality After
* Org Browser is available when connected to a scratch org.
* Retrieving from within Org Browser fixes up source-tracking correctly when connected to a scratch org.

### CODE REVIEW

**Review by randi274 — COMMENTED** [review]
I noticed while reviewing #4809 that this is the last hanger-on we have around is_scratch_org and default_username_has_no_change_tracking. Can we remove them here? 🤞 Or make a follow-up WI to remove if we have any timing concerns?

**Review by randi274 — COMMENTED** [review]
This has become an exciting cleanup PR!

QA is started, and the basics are looking good:
- Org Browser appears for scratch orgs
- Can retrieve singular file
- Refresh works 
- Can retrieve whole directories

Retesting today, and source tracking seems to be having issues for changes in remote. We'll track that issue separately.

Also regarding changes to `workspaceOrgType.ts`:
- ✅  Org Browser shows “No Default Org Set” if I log out
- ✅ Same if I refresh

File: packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts [hunk 1]
```diff
@@ -25,47 +24,28 @@ export enum OrgType {
...
   const workspaceContext = WorkspaceContext.getInstance();
-  const connection = await workspaceContext.getConnection();
+  let connection;
+  try {
+    connection = await workspaceContext.getConnection();
+  } catch (error) {
+    console.warn(
+      `An error was encountered while getting the org connection:\n ${error}`
+    );
+    return OrgType.NonSourceTracked;
```
Comment (randi274) [inline packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts:34]: If we can't return the org connection, why do we assume we have a NonSourceTracked org?

**Review by randi274 — APPROVED** [review]

File: packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts (hunk 1 above)
Comment (klewis-sfdc) [reply packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts:34]: Its basically a default value for this function to return.  When set to NonSrcTracked, commands like view changes and push/pull are hidden.

**Pull request merged**